scrapy/exporters.py

```diff
@@ -142,14 +149,15 @@ def finish_exporting(self):
         self.file.write(b"]")
 
     def export_item(self, item):
-        if self.first_item:
-            self.first_item = False
-        else:
-            self.file.write(b",")
-            self._beautify_newline()
-        itemdict = dict(self._get_serialized_fields(item))
-        data = self.encoder.encode(itemdict)
-        self.file.write(to_bytes(data, self.encoding))
+        # catch exception and don't add empty item for export
+        try:
+            itemdict = dict(self._get_serialized_fields(item))
+            data = to_bytes(self.encoder.encode(itemdict), self.encoding)
+        except TypeError as e:
+            raise e
```

Comment:
I don't think this catch+reraise does anything?

Reply:
@wRAR  thanks for your comment. 

Actually, [this](https://github.com/scrapy/scrapy/pull/5952/commits/0bd481de7cec4540aad8b9cfae896519fd9b2016#diff-b4b7b51f903343e1ac43a4655e42263010d0631f9d1702eb5058cbd5d5402d47R608) test expect the TypeError. Do you've something better in your mind for its replacement?.

Reply:
But as this code does nothing then removing it won't break the test.